src/codegen/compile.ml

```diff
@@ -1586,10 +1591,15 @@ module Tagged = struct
     | CoercionFailure -> 0xfffffffel
     | StableSeen -> 0xffffffffl
 
+  (* Declare `env` for lazy computation of the header size when the compile environment with compile flags are defined *)
+  let header_size env = 
+    if !Flags.gc_strategy == Flags.Incremental then 2l else 1l
```

Comment:
```suggestion
    if !Flags.gc_strategy = Flags.Incremental then 2l else 1l
```
use structural equality

Reply:
Thank you very much. I fixed this (including the other cases).